File: app/java-ms/management-accounting-service/src/main/java/com/example/management/domain/FinancialAnalysisResult.java
package com.example.management.domain;

/**
 * 財務分析結果
 */
public class FinancialAnalysisResult {
    private FinancialData financialData;
    private FinancialRatios financialRatios;

    public FinancialAnalysisResult(FinancialData financialData, FinancialRatios financialRatios) {
        this.financialData = financialData;
        this.financialRatios = financialRatios;
    }

    // Getters
    public FinancialData getFinancialData() {
        return financialData;
    }

    public FinancialRatios getFinancialRatios() {
        return financialRatios;
    }
}

File: app/csharp-ms/src/ManagementAccounting/ManagementAccounting.Domain/Entities/FinancialAnalysisResult.cs
namespace ManagementAccounting.Domain.Entities;

/// <summary>
/// 財務分析結果（管理会計コンテキストのドメインモデル）
/// </summary>
public record FinancialAnalysisResult(
    FinancialData Data,
    FinancialRatios Ratios,
    DateTime AnalyzedAt
)
{
    public FinancialAnalysisResult(FinancialData data, FinancialRatios ratios)
        : this(data, ratios, DateTime.UtcNow)
    {
    }
}
